backend/api/api_deployment_views.py

```diff
@@ -119,9 +125,27 @@ def create(
             headers=headers,
         )
 
-
-def get_error_from_serializer(error_details: dict[str, Any]) -> Optional[str]:
-    error_key = next(iter(error_details))
-    # Get the first error message
-    error_message: str = f"{error_details[error_key][0]} : {error_key}"
-    return error_message
+    @action(detail=True, methods=["get"])
+    def download_postman_collection(
+        self, request: Request, pk: Optional[str] = None
+    ) -> Response:
+        """Custom action to fetch a single instance."""
+        instance = self.get_object()
+        active_api_key = ""
+        try:
+            api_key_inst = instance.apikey_set.filter(is_active=True).first()
+            active_api_key = api_key_inst.api_key
+        except APIKey.DoesNotExist:
+            logger.error(f"No active API key set for deployment {instance.pk}")
+            raise NoActiveAPIKeyError(deployment_name=instance.display_name)
+
+        postman_collection = PostmanCollection.create(
+            instance=instance, api_key=active_api_key
+        )
```

Comment:
NIT: I think It's better to do this functionality in the postman_collection directory. within some service or helper class

Reply:
I considered that briefly, however having the instance creation within the class is more pythonic - I've noticed libraries follow this and it also encapsulates the functions better rather than having a standalone helper outside.
I'd like us to break away from that practice wherever possible